src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -5146,27 +5146,27 @@ def create_service_account(
                 already exists.
         """
         with Session(self.engine) as session:
+            # Create the service account
+            new_account = UserSchema.from_service_account_request(
+                service_account
+            )
+            session.add(new_account)
+
             # Check if a service account with the given name already
             # exists
-            try:
-                self._get_account_schema(
-                    service_account.name, session=session, service_account=True
+            service_accounts = session.execute(
+                select(UserSchema).where(
+                    UserSchema.name == service_account.name
                 )
+            ).fetchall()
+            if len(service_accounts) == 1:
+                session.commit()
+            else:
```

Comment:
Cannot agree in full: with current logic, I'm adding a new entity to the session and then checking if only one such entity exists as close to the commit as possible. The idea is: if some other process created the same entity in between I added it here and uniqueness check - it will prevent it. Here main assumptions are: check as close to the commit as possible and do all heavy logic beforehand, so the time between check and commit is minimal. Objections?

Reply:
Thanks, not feasible to do DB-level for now, but we will consider this.